### PR TITLE
First crack at some client side unit tests

### PR DESCRIPTION
**Pull request by dotproto**
:warning: It feels like I'm doing this wrong. I'd greatly appreciate some feedback :heartpulse:

### CODE REVIEW

**Comment by kelsin** [comment]
It looks like a full file is commented at the end of the edited file? Can we remove those comments?

**Comment by dotproto** [comment]
You must have made that comment while I was fixing it ;)

EDIT: Apparently I didn't actually fix it when I force pushed. I just changed comment text. Next commit has the fix.

**Comment by dotproto** [comment]
There are two things that feel wrong about the tests I wrote.

First, the structuring of the defines feels a bit weird. My experience with (T/B)DD is limited, but defines like "Basic configuration" or "RouteParams integration" don't seem to be the norm. Its also a bit weird to me that these defines shift the context of `it('should ...')` statements by potentially shifting the context of the test from the Controller specified in a parent define or the specific module explicitly mentioned in the parent define.

Second, it feels super repetitive to create these `beforeEach( ... )` blocks inside the most specific defines. I started doing this because I wanted to ensure that the controller was [initializing with an empty array](https://github.com/legitco/gimli/pull/65/files#diff-633bc4c166dc63349552be73b3db0c86R38). I tried reading through some Angular docs and sites discussing testing with AngularJS, but a good chunk of the highly technical resources were out of my depth. My biggest failing in trying to understand these docs was my lack of experience with testing -- the ideas just didn't click.

**Comment by dotproto** [comment]
Below are a couple resources I came across that may be worth further discussion.
- https://docs.angularjs.org/guide/unit-testing
- https://docs.angularjs.org/api/ngMock/function/angular.mock.inject
- https://docs.angularjs.org/api/auto/object/$provide
- https://docs.angularjs.org/api/ngResource/service/$resource
- http://stackoverflow.com/questions/19297258/why-is-provide-only-available-in-the-angular-mock-module-function-and-q-onl

File: test/client/clientSpec.js [hunk 1]
```diff
@@ -1,28 +1,80 @@
...
-      expect(scope.issues.length).to.be.empty;
-    });
+          ctrl = $controller('IssuesController', {
+            $scope: scope
+          });
+        })); // beforeEach
+
+        it('should exist', function () {
+          expect(ctrl).to.exist;
+          expect(ctrl).to.not.be.undefied;
```
Comment (mschnee) [inline test/client/clientSpec.js]: undefied?  undefined?

**Comment by dotproto** [comment]
I'm going to plow forward with the iterative approach. Some tests are better than no tests. When we realize that this PR is filled with terrible things we can replace my work with something better. I'll merg this as soon as Travis gives the thumbs up.

**Pull request merged**